### PR TITLE
git cherry-pick enhancement

### PR DESCRIPTION
**Pull request by koecher**
This PR is an enhancement to candi to allow for cherry-picks of development commits to a stable release.

The reason is, that a more or less long time undergoes between two stable releases within a package and sometimes there is the need (for a candi-user or from our developer side) to select a very little number of (development) commits added to a stable release without using the complete unstable development version.

This PR includes an example for a bugfix of a deal.II class (disabled by default), but it can be used for other packages which we check out with git.

For future use, maybe problems as in PR #92 can be easily resolved with that within candi.

I barely checked that this PR does not break anything in candi, i.e. initial build experience for deal.II v9.0.0 for fedora 28. Further tests might be required.

### CODE REVIEW

File: candi.sh [hunk 1]
```diff
@@ -349,6 +349,16 @@ package_fetch () {
             cd ${EXTRACTSTO}
             git checkout ${VERSION} --force
             quit_if_fail "candi: git checkout ${VERSION} --force failed"
+            cd ..
+        fi
+        
+        # git cherry-pick on commits given by ${CHERRYPICKCOMMITS}
+        if [ ${STABLE_BUILD} = true ] && [ ! -z "${CHERRYPICKCOMMITS}" ]; then
```
Comment (koecher) [inline candi.sh]: Note: I'm not sure if we should place the application of git cherry-pick after "unpacking" in the git trunc.
Another (and may be better opportunity) would be to place it under `apply_(specific)_patches()`, but this function might not be implemented by default.
Any comment on this is helpful for the discussion.

**Comment by tjhei** [comment]
- I don't think this is likely to break anything.
- I think doing this after unpacking makes sense.

Therefore, I am okay with merging as is. That said, I don't think it is a very useful feature as we can already apply patches for packages and I doubt people want to often apply commits for deal.II (and if they apply cleanly...).

**Comment by koecher** [comment]
Indeed, only a (very) little of candi-users might need this feature.

On the other hand, currently our complete group need one or a few bugfix commits of deal.II, but not the complete (unstable) development branch, to make their software running as expected. This can happen more often as one thinks of.

I would not claim, that this choice is easy, since one needs to know if the application of the commit is working.

Tomorrow, I'll update the PR for moving the cherry-pick to the `apply_patch` function of candi.

**Comment by tjhei** [comment]
> This can happen more often as one thinks of.

In such a situation I would probably create my own branch under github.com/tjhei/dealii and use that instead of the official repo. That means you can cherry-pick and manually fix things.

**Comment by koecher** [comment]
Ups. I think I have messed up 2 different discussions. Sorry.

For this PR and your last answer:

Well, thats correct Timo.

But sometimes it is very useful to have certain commits added to a stable release. Those should be selected by special "software-engineering" guys of the group, I suppose.

For instance, we had under our current developments in my software two issues with deal.II (obvious programming errors) which are not reflected by v9.0.0 and will not be reflected by v9.0.1. But on the other hand, this software (must be) is used by non-software guys, which we want to give only candi and a `local.cfg` which will work. This is my idea behind this PR.

**Comment by masterleinad** [comment]
> For instance, we had under our current developments in my software two issues with deal.II (obvious programming errors) which are not reflected by v9.0.0 and will not be reflected by v9.0.1

Then, we like to hear about them as soon as a release is tested. :wink:

**Comment by koecher** [comment]
@masterleinad those issues / PR are already merged into master, but not in v9.0.1 :-)

**Comment by koecher** [comment]
I've pushed a commit which moves the git cherry-pick to the "patch" location. From my side this is ready to merge, if everyone else is satisfied as well.

**Pull request merged**